applications/nrf_desktop/src/util/config_channel.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef _CONFIG_CHANNEL_H_
+#define _CONFIG_CHANNEL_H_
+
+#include "config_event.h"
+
+#define CONFIG_CHANNEL_FETCHED_DATA_MAX_SIZE 4
+
+struct config_channel_frame {
+	u8_t report_id;
+	u16_t recipient;
+	u8_t event_id;
+	u8_t status;
+	u8_t event_data_len;
+	u8_t *event_data;
+};
+
+struct fetch {
+	atomic_t done;
+	u8_t data[CONFIG_CHANNEL_FETCHED_DATA_MAX_SIZE];
+
+	/* Recently fetched event_id and recipient. */
+	u8_t event_id;
+	u16_t recipient;
+};
+
+struct config_channel_state {
+	atomic_t status;
+	bool transaction_active;
+	bool disconnected;
+	bool is_fetch;
+	struct k_delayed_work timeout;
+	struct config_channel_frame frame;
+
+	struct fetch fetch;
+};
+
+void config_channel_init(struct config_channel_state *cfg_chan);
```

Comment:
Should we document the functions in this header file?

Reply:
I will create documentation for this module in subsequent pull request.